### ISSUE TITLE
Add options to provide custom fallback value for source generators

### ISSUE THREAD

**Issue by Ashymonth** [sveinungf/spreadcheetah#67]
I'd like to have the ability to provide my own custom default value. For example, I need to make a report. I have a rule that if a number is null, then I need to fill the cell with a "-" character. But currently, I can't do it. If I set my number property to string, then in the Excel file it will also be a string. So I can't have all options for numbers that Excel has. But if my property is Nullable<number> then I can't provide "-" character for it.

**Comment by sveinungf** [comment]
You're right, there is no way to do this right now when using the source generator. To do this right now you will have to map your type to `DataCell`s manually and then call `AddRowAsync`.

For the source generator I'm considering how to offer more flexibility on how properties are mapped to cells. What I currently have in mind is to be able to define a mapping class with a specific shape, and then instruct the source generator to use it by using an attribute. For example like this:

```csharp
public class MyObject
{
    [CellValueMapper(typeof(NullToDashMapper))]
    public int? Number { get; set; }
}

public class NullToDashMapper : ICellValueMapper<int?>
{
    public DataCell MapToCell(int? value)
    {
        return value is null ? new DataCell("-") : new DataCell(value.Value);
    }
}
```
This is not possible right now, but I think that could solve your use case and also many other possible use cases.

**Comment by Ashymonth** [comment]
What do you think about providing style in this way? If I need to add style to a cell, I need to pass the ```StyleId``` for it. Currently, there is no public API to get a style. Should we add an API to get the ```StyleId``` from the SpreadSheet and create an interface method like ```public StyledCell MapToCell(SpreadSheet sheet, int? value)```? Or something different?

**Comment by sveinungf** [comment]
I'm currently working on #44 which will add support for named styles, and also a public API to get a style by its name. A mapping class like I suggested above could potentially handle styling by referencing style names. Having access to the `Spreadsheet` instance in the mapping class is something I would like to avoid, because then you could call methods that I think we all agree should never be called in that context.

**Comment by sveinungf** [comment]
This has been implemented in version 1.17.0.
